### PR TITLE
feat: add text component & update renderer methods

### PR DESCRIPTION
**Pull request by TekMath**

### CODE REVIEW

**Review by Yann-Masson — DISMISSED** [review]
If we had more time I would have asserted my authority by refusing this PR for some triviality but we're not on time.

File: games/components/ITextureComponent.hpp [hunk 1]
```diff
@@ -0,0 +1,38 @@
+/*
+** EPITECH PROJECT, 2024
+** arcade-shared [WSL: Ubuntu-22.04]
```
Comment (flavien-chenu) [inline games/components/ITextureComponent.hpp]:
```suggestion
** arcade-shared
```

File: games/components/IDisplayableComponent.hpp [hunk 2]
```diff
@@ -2,7 +2,7 @@
 ** EPITECH PROJECT, 2024
 ** arcade-shared [WSL: Ubuntu-22.04]
```
Comment (flavien-chenu) [inline games/components/IDisplayableComponent.hpp]:
```suggestion
** arcade-shared
```

File: graphics/IWindow.hpp [hunk 3]
```diff
@@ -95,11 +96,18 @@ class shared::graphics::IWindow {
...
 
     /**
-     * @brief Render the entity with given properties
+     * @brief Render the texture of entity with given properties
      *
-     * @param props Properties of the entity to render
+     * @param props Properties of the entity & texture to render
      */
-    virtual void render(const EntityProps &props) = 0;
+    virtual void renderTexture(const TextureProps &props) = 0;
```
Comment (flavien-chenu) [inline graphics/IWindow.hpp]:
```suggestion
    virtual void render(const TextureProps &props) = 0;
```

File: graphics/IWindow.hpp [hunk 4]
```diff
@@ -95,11 +96,18 @@ class shared::graphics::IWindow {
...
+
+    /**
+     * @brief Render the text of entity with given properties
+     *
+     * @param props Properties of the entity & text to render
+     */
+    virtual void renderText(const TextProps &props) = 0;
```
Comment (flavien-chenu) [inline graphics/IWindow.hpp]:
```suggestion
    virtual void render(const TextProps &props) = 0;
```

**Review by flavien-chenu — CHANGES_REQUESTED** [review]
Non

**Review by AxelF44 — DISMISSED** [review]

**Review by flavien-chenu — APPROVED** [review]
Oui

**Review by Yann-Masson — APPROVED** [review]

**Review by AxelF44 — APPROVED** [review]

**Pull request merged**